mpstat.c

```diff
@@ -2211,6 +2221,58 @@ void rw_mpstat_loop(int dis_hdr, int rows)
 	}
 }
 
+/*
+ ***************************************************************************
+ * Detect physical vCPU hotplug and call rw_mpstat_loop to display stats
+ *
+ * IN:
+ * @dis_hdr	Set to TRUE if the header line must always be printed.
+ * @rows	Number of rows of screen.
+ ***************************************************************************
+ */
+void mpstat_loop(int dis_hdr, int rows)
+{
+	while (1) {
+		photplug = FALSE;
+		/* Main loop for reading-writing stats */
+		rw_mpstat_loop(dis_hdr, rows);
+
+		/* handle vCPU physical hotplug */
+		if (photplug == FALSE) {
+			break;
+		}
+		else if (photplug == TRUE) {
+			/* Update the highest processor number */
+			cpu_nr = get_cpu_nr(~0, TRUE);
+
+		        /* Recalculate number of interrupts per processor */
+		        irqcpu_nr = get_irqcpu_nr(INTERRUPTS, NR_IRQS, cpu_nr) +
+					NR_IRQCPU_PREALLOC;
+		        /* Recalculate number of soft interrupts per processor */
+			softirqcpu_nr = get_irqcpu_nr(SOFTIRQS, NR_IRQS, cpu_nr) +
+					NR_IRQCPU_PREALLOC;
+
+			/*
+			 * Reallocate cpu stats structs :
+			 * global, proc0, proc1, ..., proc$(prev_cpu_nr-1).
+			 * global, proc0, proc1, ..., proc$(prev_cpu_nr-1), ..., proc$(cpu_nr-1).
+			 */
+			realloc_mp_struct(cpu_nr + 1);
+
+			/* Get NUMA node placement */
+			node_nr = get_node_placement(cpu_nr, cpu_per_node, cpu2node);
+
+			/*
+			 * Don't buffer data if redirected to a pipe.
+			 * Note: With musl-c, the behavior of this function is undefined except
+			 * when it is the first operation on the stream.
+			 */
+			setbuf(stdout, NULL);
```

Comment:
Hmm... Not sure if this is still needed here since `setbuf()` has already been done in `main()` => remove it.

Reply:
yes, removed this